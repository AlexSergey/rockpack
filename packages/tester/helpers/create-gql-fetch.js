const fetch = require('isomorphic-fetch');

const createGqlFetch = (url) => async (query) => {
  try {
    const response = await fetch(url, {
      body: JSON.stringify(query),
      headers: {
        'Content-Type': 'application/json',
      },
      method: 'POST',
    });
    if (!response.ok) {
      throw new Error(response.statusText);
    }

    return await response.json();
  } catch (e) {
    throw new Error(e);
  }
};

module.exports = createGqlFetch;
