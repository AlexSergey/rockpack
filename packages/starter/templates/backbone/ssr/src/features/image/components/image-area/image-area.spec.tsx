import renderer, { act } from 'react-test-renderer';

import { createAppWrapper } from '../../../../tests/create-app-wrapper';
import { sleep } from '../../../../tests/helpers';
import { ImageArea } from './index';

it('ImageArea renders correctly', async () => {
  const AppWrapper = createAppWrapper();
  const imageArea = renderer.create(
    <AppWrapper>
      <ImageArea />
    </AppWrapper>,
  );

  await act(sleep(100));

  const tree = imageArea.toJSON();

  expect(tree).toMatchSnapshot();
});
