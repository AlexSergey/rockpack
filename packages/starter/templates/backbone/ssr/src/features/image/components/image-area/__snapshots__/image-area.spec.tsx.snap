// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageArea renders correctly 1`] = `
<div
  className="image-holder"
>
  <div
    className="img"
  >
    <img
      alt="random"
      src="https://picsum.photos/id/0/5616/3744"
      width="200px"
    />
  </div>
</div>
`;
